src/main/java/com/mparticle/kits/ButtonKit.java

```diff
@@ -50,20 +39,33 @@ public ButtonKit getInstance() {
     }
 
     @Override
-    protected List<ReportingMessage> onKitCreate(Map<String, String> settings, Context context) {
+    protected List<ReportingMessage> onKitCreate(Map<String, String> settings, final Context ctx) {
         final String applicationId = settings.get("application_id");
         if (KitUtils.isEmpty(applicationId)) {
-
-            throw new IllegalArgumentException("No Button application ID provided, can't initialize kit.");
+            throw new IllegalArgumentException(
+                    "No Button application ID provided, can't initialize kit.");
         }
+        ButtonMerchant.configure(ctx, applicationId);
```

Comment:
This context (and any via the `getContext()` method) is managed by the mParticle Core SDK itself. I'd lean on the assumption that Core is handling 'lifecycle' properly. And since this isn't an Activity, the application context isn't available. Unless you meant to extract it from the passed context itself?